### ISSUE TITLE
data time issue

### ISSUE THREAD

**Issue by a30000329** [amcharts/amcharts4#2277]
Please use [this link](https://codepen.io/alexfan/pen/BayvVgB?editors=0011) for issue recreation. Issues found:

![image](https://user-images.githubusercontent.com/46702190/77119613-ee0bc000-6a8a-11ea-9ca2-31b84adffd48.png)

- using above screenshot, the 13% data point is 7am, why does it display 6am?
![image](https://user-images.githubusercontent.com/46702190/77039648-80b84a80-6a0a-11ea-937c-7c04e0b78ede.png)

We also found if reduce the iso date time string last section from for example "2020-01-20T07:00:00.00000000Z" to "2020-01-20T07:00:00.000Z", this issue is gone

**Comment by a30000329** [comment]
@martynasma separate issue created. thanks

**Comment by zeroin** [comment]
If you do this:
console.log(new Date("2020-01-20T04:00:00.000Z"))

You will see that the console outputs date at your time zone. 
Setting `chart.dateFormatter.timezoneOffset = 0;` should fix the problem.

**Comment by a30000329** [comment]
@zeroin 
please note that this will not fix the problem:
![image](https://user-images.githubusercontent.com/46702190/77419196-20674580-6e1c-11ea-894c-0f2a8272ebe0.png)

https://codepen.io/alexfan/pen/qBEQZoy?editors=0010

**Comment by martynasma** [comment]
The date format `"i"` expects three digits for milliseconds, whereas you have 9:

```
"2020-01-20T22:00:00.000000000Z"
```

Try formatting your dates like this:

```
"2020-01-20T22:00:00.000Z"
```

It will format dates properly.

**Comment by a30000329** [comment]
thanks for the reply. so there is no plan to support more precisions? the reason is normally web API like .net web API will return 6 digits for iso date-time string. So I think it will be helpful if you can support this.

**Comment by martynasma** [comment]
Well, we do not support smaller precision than millisecond. I don't think even JavaScript supports smaller precision.

That said, it probably would not hurt to at least not fail of there are more digits.

I think we can make it happen. Will keep you posted.

**Comment by a30000329** [comment]
thanks for your update. I appreciate that you think the same way. The current behavior is a bit unexpected.

**Comment by martynasma** [comment]
Fixed in [4.9.12](https://github.com/amcharts/amcharts4/releases/tag/4.9.12).

## [4.9.12] - 2020-03-26

### Added
- New mouse cursor style added: `am4core.MouseCursorStyle.text`.
- Export `formatOptions` now accessible via public property, e.g. `chart.exporting.formatOptions.getKey("csv").disabled = true`.

### Changed
- Using date format `"i"` to parse ISO dates now supports unlimited number of milliseconds. It was failing before if more than three digits were used for milliseconds.

### Fixed
- If `Label` had its interactions disabled then re-enabled, its `selectable` property was being ignored.
- `Modal` with `closable = false` could still be closed via curtain click.
- `openModal()` ignored title (second parameter).
- Pushing child elements in `Label` which has `textValign` set was resulting in error.
- `cursor.snapToSeries` was not working properly with `CategoryAxis`.
- Firefox was not measuring labels properly if a chart was lazy-loaded using `onlyShowOnViewport`.

[Full change log](https://github.com/amcharts/amcharts4/blob/master/dist/es2015/CHANGELOG.md).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.

**Comment by a30000329** [comment]
Legend! thanks a lot!!
